grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/rule/InferenceRule.java

```diff
@@ -81,14 +82,9 @@ private void propagateConstraints(Atom parentAtom){
             Set<Atom> types = parentAtom.getTypeConstraints().stream()
                     .filter(type -> !body.containsEquivalentAtom(type))
                     .collect(Collectors.toSet());
-            Set<Predicate> predicates = new HashSet<>();
-            //predicates obtained from types
-            types.stream().map(type -> (Binary) type)
-                    .filter(type -> type.getPredicate() != null)
-                    .map(Binary::getPredicate)
-                    .forEach(predicates::add);
-            //direct predicates
-            predicates.addAll(parentAtom.getPredicates());
+            Set<Predicate> predicates = parentAtom.getPredicates();
+            if(parentAtom.isResource())
+                    predicates.addAll(((Resource) parentAtom).getMultiPredicate());
```

Comment:
Whoops! Didn't realise this isn't a graph API `Resource` object, but a reasoner class.